### ISSUE TITLE
Still need to force quit 2.4.0b4 on 10.10.1

### ISSUE THREAD

**Issue by bobinchicago** [dscottbuch/cTiVo#83]
Hi there,
 
I'm using 2.4.0b4 on 10.10.1 -- I'd been using the last release version just fine under 10.10.0. 
 
There are two behaviors. In the first and more common, cTivo retrieves the first (most recent) 15 items on my Tivo (Series 3 HD, 11.0m-01-2-652) and then shows the spinning cursor. The Force Quit dialog confirms the program isn't responding and offers the force quit menu item. I have seven subscriptions, some of which are daily and some of which are inactive, but that 15th item is not always a subscribed program. The Tivo is on my Ethernet network.
 
The second is that the program retrieves the full recorded show list (I can only confirm to the bottom of that field, but it goes past the 15 most recent, anyway) and then dies a few seconds later, but again, I don't have any clue what triggers it; the best I can offer is that this seems to happen when I haven't yet interacted with the program. I can't vouch for that consistently. 
 
Either way, if I keep the Force Quit dialog open in the background, cTivo changes from black text to red Not Responding flag in what seems to be about the same duration of time.

Since it takes a few moments to crash, I was finally able to reach the log menu in the Preferences pane and change it to verbose. If I can put that anywhere, I'm happy to do that. It took many tries to hit that in the moments before the crash and I doubt I can set it to any other level of granularity reliably. And if anyone has any troubleshooting suggestions, I'm happy to try them and report back. I've already tried deleting the cTivo.plist file and relaunching, to no avail. Thanks!

**Comment by bobinchicago** [comment]
Adding a third subscription went fine. After everything was downloaded and encoded, adding a fourth subscription with nine episodes (mostly partial; I occasionally save a particularly goofy segment of Cash Cab) crashed but without the beach ball -- Activity Monitor showed CPU usage spiking to 20% and generally in the 10-12% range. The cursor was displayed but there was no response to clicks. After I ran some errands, it was still in that state a couple of hours later, so I force quit, restarted, and added a subscription with one episode; that went just fine. Hopefully the high level of logging helps sort it out; log is linked. Thanks again for all your efforts, and happy new year -- https://www.dropbox.com/s/ns84ieeirkq3ghw/com.cTiVo.cTiVo%202014-12-31%2020-21.log?dl=0

**Comment by stopcollaboratelisten** [comment]
Hey Guys. I downloaded the new alpha, and it worked great for the first couple of days, but then I don't know what changed. I could not get the app to do anything except spin the beach ball. I even downloaded 2.3 and the same thing happened. Well, I found this thread and downloaded the cTivo super-logger app and fired it up. It seems to be working well now. It picked up the subscriptions I already had and started working on them right away. I will keep you updated, even if this thread closes.

**Comment by bobinchicago** [comment]
Hm. I haven't seen the original problems I posted about in some time, but the problem in which I select any episode of Cash Cab and get a crash when I click Subscribe is repeatable (downright persistent, in fact). If you want me to open another ticket here for that for the sake of tracking, let me know, and by the same token, if there are any other tests you'd like me to conduct, I'm happy to do so. 
 
I can download the CC episodes just fine by selecting each and clicking Download, but clicking Subscribe gives me an immediate stop, not even the beach ball: The button stays highlighted, fields don't scroll, and menus don't appear. Also, after force quitting and relaunching, cTivo couldn't find my Tivo; both my Mac Mini and my Tivo are connected by Ethernet, although the Mac's hooked to the router itself and the Tivo's linked to an 8-port switch in the living room. After another (normal) quit and relaunch, cTivo found my Tivo just fine.
 
The log's at https://www.dropbox.com/s/wvxxc59gghrgsl2/com.cTiVo.cTiVo%202015-01-04%2019-49.log?dl=0 -- again, I'm happy to open another ticket if you have a sense that this is a different bug, and if there's any testing you'd like me to try, don't hesitate to ask. Thanks again for your efforts. (And thanks for the extra data point, stopcollaboratelisten.)

**Comment by mackworth** [comment]
Sigh.  I'm pouring time into the beachball issue, but haven't found it yet.  Some good clues, but no resolution.  

This one looks different.  I actually need your Console logs here. If you can open `All messages` and look for any messages between the last msg from cTiVo and the start of the next run:
`2015-01-04 17:56:39:442`  to  `2015-01-04 18:00:09:212`

@ stopcollaboratelisten, I can't say I recommend using the logging version as your production one. Have you tried running the beta4 again after the logging version worked?

**Comment by bobinchicago** [comment]
15K text file waiting for you at https://www.dropbox.com/s/f99yav6mm0ps8ll/cTivo%20all%20messages%20from%20console.txt?dl=0

**Comment by bobinchicago** [comment]
In the meantime, if any of those clues you mention on the beachball issue suggest anything you'd like me to try at my end, let me know. If the only difference between 2.4.0 beta 4 and the logging version you provided is the amount of logging, it's kind of odd that I haven't seen the beachball since installing the logging version, so I don't mind poking around. Unfortunately, I didn't end up saving any of those old prefs files, but it's a small sandbox to play in if you're looking at any of the prefs.

**Comment by mackworth** [comment]
Ok, looks like all kinds of things started happening at 5:57:45. The direct cause of cTivo being terminated is that the system accused it of "burning CPU! It used more than 50% CPU over 180 seconds", which is certainly accurate given the nature of what we do, but hardly a good reason to terminate it; haven't seen that before.

Secondly, and possibly related, there's all kinds of other stuff happening with (a)your Buffalo NASNavigator (it keeps trying to run, but quits after 11-13 seconds), and (b and probably more relevant) running into memory pressure (memoryStatus_thread), which is killing a bunch of system processes. From this brief snapshot, I'd say you're running too many apps, or just need to reboot and clean things up.

So, the only thing surprising here is that you say it's repeatable. Obviously the system wouldn't terminate for excess CPU usage if you haven't been previously downloading.  Can you do another run without any downloads previously and send the relevant log and console section?

**Comment by mackworth** [comment]
Hadn't tried before your edit, but URL now says "The file you're looking for has been moved or deleted"

**Comment by bobinchicago** [comment]
Bizarre. On my laptop, I clicked and it downloaded -- I did some other work in Dropbox tonight and maybe it was just a transient issue. Sorry for the glitch. As it is, it's in my Public folder there and it's just the zip file and its original components, if that works better.
 
https://dl.dropboxusercontent.com/u/3024834/cTivo%20debugging%202014-01-05.zip
 
https://www.dropbox.com/sh/s2n1finf653s15i/AAC5Gyiok-n_fhwpIBScBacLa?dl=0

**Comment by ryan-orr** [comment]
I'm running into this issue as well. Are there logs I can provide?

**Comment by bobinchicago** [comment]
In the meantime, while I haven't seen the beachball since installing the heavy logging version, I just had another hang when adding a different subscription, and I might have something for you. You'd mentioned a suspicion that it had something to do with field sizes earlier. I had six subscriptions and had the Subscriptions field set to show about 6 1/2 lines. Once I expanded the subscriptions field to about nine lines and added the subscription, everything went as it should -- the first episode is downloading and the second is queued. This zip file has three logs: the first of the state of the application before I did anything, the second is after the hang, and the third is after the successful subscription addition. 
 
https://dl.dropboxusercontent.com/u/3024834/2014-01-11%20915pm.zip
 
If there are any glitches with that file as there were with others, my public Dropbox folder should be at
 
 https://www.dropbox.com/sh/s2n1finf653s15i/AAC5Gyiok-n_fhwpIBScBacLa?dl=0 
 
And, as always, if you'd like me to test anything at this end, please let me know. Thanks!

**Comment by mackworth** [comment]
All: Ok, I still can't reproduce the beachball symptom on my system, but I've spent many hours pouring through what might be causing it.  Here's a new version (beta 6 (520); without heavy logging) that restructures how I present the download and subscription tables.  It's taken a more mainstream approach to some aspects and has cleaned up a couple others, although no Aha! moment that would certainly cause the problem.  So please give it a try and let me know if it still has the problem for you.

https://dl.dropboxusercontent.com/u/21507587/cTiVo.zip

**Comment by bobinchicago** [comment]
Thanks. I threw all of my previous downloads into a separate folder, cleared subscriptions and downloads, and launched without a problem. I'll add subscriptions and let them download until either something happens or nothing does. Will keep you posted.

**Comment by bobinchicago** [comment]
Although I'm still not seeing the beachball, I'm afraid I still have the hang when adding a subscription that would display below the bottom of the field. Enlarging the field so that there are empty lines and then subscribing is no problem. The zip file linked here has five logs in the folder -- the first three are the logs that were there when it hung; the fourth reproduces the hang; and the fifth is after I enlarged the field and subscribed.
 
 https://dl.dropboxusercontent.com/u/3024834/2015-01-14%20cTivo%20hang%20logs.zip
 
 And there's more this time: The previous subscription was to Cash Cab; I think I've mentioned that I have a number of episodes and show fragments, and 7 of 9 episodes failed. At first I thought the 7 failed downloads were the fragments, but it didn't turn out that way. Details on this will be in the earliest of the logs. 
 
 As always, if I can test or check anything for you, I'm happy to.

**Comment by mackworth** [comment]
Ok, unfortunately the logs aren't telling me much. They just stop in the hang case and keeps going in the other one.

So first, I'm hearing that this is completely reproducible? If you have no empty lines in the subscription table, it hangs every time, and when you do, it doesn't?  Same show (as I see you did), and it doesn't matter what show you use?

Second, does it happen when the queue is paused (Cmd-U)?

**Comment by bobinchicago** [comment]
Sorry for the delay, and sorry the logs haven't been more useful. But yes, it's reliably reproducible and you describe it accurately. 
![will hang](https://cloud.githubusercontent.com/assets/10276192/5793433/f9083478-9f08-11e4-9535-54d555346ad9.png) consistently hangs and 
![won t hang](https://cloud.githubusercontent.com/assets/10276192/5793440/15ad5c98-9f09-11e4-85f3-b8e91d5e7517.png) doesn't. It happens whether the queue is paused or not and whatever the show happens to be. 
 
 If you think there's the chance of this capturing any useful info, I'd be happy to grow that box a pixel at a time (it's not like there are that many -- a commercial break should be plenty of time) and find the trigger point. But by the same token, if I'm the only report you've had of this particular hang, maybe it's time to invoke "Doctor, it hurts when I do this." "So don't do that." and close the thread as something that's affecting only one user who found an easy workaround.

**Comment by mackworth** [comment]
And this is with the beta 6 (version 520 or later)? And by hang, the beachball eventually comes up?

No, I'd really like to see if I can find this. It's an odd one, and given that others have seen similar symptoms...

I have a theory that your prefs file has something strange in the table-saving section.

Fastest way to test is to reset to default preferences, but we'll save the current one to restore back...
0) exit cTiVo
1) Go to your preferences folder: Finder>Go>Go To Folder>~/Library/Preferences/` 
1b) Wait about five to ten seconds for the preferences file to be written out.
2) rename `com.cTiVo.cTiVo.plist` to `com.cTiVo.save.plist` 
3) run Activity Monitor
4) click on the process `cfpresd` under your User name (not `root` or `administrator`). Note the PID number.
5) hit the little x in the upper left corner, and hit Quit. 
It should disappear from the list and then quickly reappear but with a different PID number.
6) Run cTivo again. Everything should be back to defaults, no queue, no subscriptions...
7) Now add subscriptions to try your scenario again.

If it still fails, then just follow all the same steps again, but rename the `save.plist` back to `com.cTiVo.cTiVo.plist` again. Next I'll need to do more specific logging to see what's going on.

If it doesn't show the symptom, then you can also put back the plist, but post me a copy of it. I'll edit out the table section and send it back to you to see if that's the particular problem within the file.

**Comment by bobinchicago** [comment]
Sorry for my absence -- was down with bronchitis. Immediately, the only answer I have is that I haven't seen the beachball since the alphas and betas you've provided -- the program stops responding but the cursor remains the arrow in these versions in this thread. The last time I saw the beachball was in the last stable public release. In the meantime, I'll follow your testing steps over this weekend and report back.

**Comment by mackworth** [comment]
Any progress on this with 2.4.0beta 7?

**Comment by bobinchicago** [comment]
I'm sorry for my prolonged absence -- I've been running cTivo on a Mac Mini that I use as a media server; a while back, I started having intermittent problems with it on the network -- I managed it via Screen Sharing and that in particular became less and less reliable -- and within the last couple of days, whoever hacked it changed my password and locked me out. Even a power cycle and command-R to reach the recovery screen tells me that the home directory is invalid (I'm no longer the administrator account) and gives me a login window that rejects my login info. I apologize for the timing -- I've got an appointment with the Genius Bar after work tomorrow, but I don't know whether they'll be able to restore it or have me swap drives and start me from scratch or whatever else. Either way, I also need to find some solid info on hardening my network before getting it back up and running and find out why Yosemite on the Mac Mini's so vulnerable. Once I have a plan, I'll swing back to let you know when I can pick this back up -- this is pretty vexing and I don't want it to take too long. Hopefully, if I have to reinstall, this bug will persist. *That* would be ironic...

**Comment by mackworth** [comment]
So sorry to hear that; what a pain. Hope things settle out for you soon!

**Comment by bobinchicago** [comment]
They did a great job and nearly everything was saved. Because they created a new account and moved everything from one user folder to another, I'm still having some related issues, mostly permissions but occasionally just eccentric behavior. It's mostly shaken out but since the NAS I've saved stuff to is part of the problem, I'll give it a couple more days of use and then pick up where I left off. 
 
Checking my apps folder, I have versions 2.3.1 (the last stable release, I think?), 2.4.0 b5, and 2.4.0 b6, but I see you mentioned a b7 -- is that available at your Dropbox link? Is there any value in keeping the old ones around?

**Comment by mackworth** [comment]
there's now a beta 7 in the releases. No, I'd get rid of all the older releases, except if 2.3.1 works and 2.4.0b7 fails, then that would be critical information. I'm eager to release 2.4.0 as a stable release, but I don't know if the Yosemite issue was introduced by 2.4.0 or not. (Crashlytics crash tracking was added in 2.4.0, so I don't know if this issue was there before).

**Comment by bobinchicago** [comment]
Well, yay: No crash on b7. The only step I added to your list above was that, after it refreshed the show list from the Tivo but before I did anything else, I changed the format to Handbrake TV. I left the app window and the list fields within at their default size.
 
Unfortunately, one of the downsides of the permissions errors I mentioned above is that both Dropbox and Google Drive have become disconnected. Apparently this is so common for Dropbox that it popped up a dialog with the troubleshooting page, and although I need to get moving this afternoon, I'll curl up with that tonight so I can post the plist files to you. But I did want to let you know that so far, so good with b7.

**Comment by bobinchicago** [comment]
With that done, the original (crashing) plist file is at 
 
https://www.dropbox.com/s/b9hkadnne1v2ful/com.cTiVo.save.plist?dl=0
 
and you didn't mention wanting the new one, but if there's any value to it, no reason not to:
 
 https://www.dropbox.com/s/ddl9ive9faeaydd/com.cTiVo.cTiVo.plist?dl=0

**Comment by mackworth** [comment]
So the key for me is a reproducible case. If the old one still crashes in 7 and the new one doesn't, that would be very useful!

**Comment by bobinchicago** [comment]
So while b7 was not running I renamed the fresh new plist, renamed the original (b6 and before) back to cTivo, quit the cfprefsd process and waited for it to relaunch, launched b7, saw that the window position and list field dimensions were what they used to be and watched the queue and particularly the subscription lists populate, watched it start to download the first program in the queue, and added two subscriptions without incident. Woot.
 
 The zip file at https://www.dropbox.com/s/z2w5lknmy50kt39/cTivo%20plists.zip?dl=0 includes the log file from tonight (dunno if there's any value but it doesn't hurt to throw it in) and three plist files -- a Finder duplicate of the b6 and before plist file, the plist file that b7 created freshly, and the b6 file that I used to test b7. There's also a text file that says that last sentence perhaps a bit more coherently. 
 
 I'll be away this weekend, but should have some time Thursday night and Friday morning for any testing -- otherwise I'll be around again Monday night.
